File: java/src/test/java/com/secookbook/examples/chapter08/pageobjects/BmiCalcPage.java
package com.secookbook.examples.chapter08.pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.LoadableComponent;
import static org.junit.Assert.*;

public class BmiCalcPage extends LoadableComponent<BmiCalcPage> {

	private WebElement heightCMS;
	private WebElement weightKg;
	private WebElement calculate;
	private WebElement bmi;
	private WebElement bmiCategory;

	private WebDriver driver;

	private String url = "http://cookbook.seleniumacademy.com/bmicalculator.html";
	private String title = "BMI Calculator";

	public BmiCalcPage(WebDriver driver) {
		this.driver = driver;
		PageFactory.initElements(driver, this);
	}

	@Override
	protected void load() {
		this.driver.get(url);
	}

	@Override
	protected void isLoaded() throws Error {
		assertTrue("Bmi Calculator page not loaded",
				driver.getTitle().equals(title));
	}

	public void calculateBmi(String height, String weight) {
		heightCMS.sendKeys(height);
		weightKg.sendKeys(weight);
		calculate.click();
	}

	public String getBmi() {
		return bmi.getAttribute("value");
	}

	public String getBmiCategory() {
		return bmiCategory.getAttribute("value");
	}
}

File: python/chapter08/bmi_calc_page.py
class BmiCalcPage(object):
	def __init__(self, driver):
		self._driver = driver
		self._url = 'http://cookbook.seleniumacademy.com/bmicalculator.html'
		self._title = 'BMI Calculator'
	
	@property
	def is_loaded(self):
		return self._driver.title == self._title
	
	@property
	def bmi(self):
		bmi_field = self._driver.find_element_by_id('bmi')
		return bmi_field.get_attribute('value')
	
	@property
	def bmi_category(self):
		bmi_category_field = self._driver.find_element_by_id('bmi_category')
		return bmi_category_field.get_attribute('value')
	
	def open(self):
		self._driver.get(self._url)
	
	def calculate(self, height, weight):
		height_field = self._driver.find_element_by_id('heightCMS')
		weight_field = self._driver.find_element_by_id('weightKg')
		calc_button = self._driver.find_element_by_id('Calculate')
		
		height_field.send_keys(height)
		weight_field.send_keys(weight)
		calc_button.click()